commit/merkleroot/rmn/client.go

```diff
@@ -0,0 +1,19 @@
+package rmn
+
+import (
+	"context"
+	"errors"
+)
+
+// ErrTimeout is returned when the signature computation times out.
+var ErrTimeout = errors.New("signature computation timeout")
+
+// Client contains the methods required by the plugin to interact with the RMN nodes.
+type Client interface {
+	// ComputeSignatures computes and returns the signatures for the provided lane updates.
+	ComputeSignatures(
```

Comment:
Does this correspond to the `ReportSignatureRequest` in the RMN design doc?

IMO: in order to make things consistent I'd rather we have consistent naming across design docs and code. That way when RMN maintainers read our code its pretty clear what these functions are doing (and vice versa, if it ever comes to us reading RMN code)

Reply:
Not really since it encapsulates both `ObservationReq` and `ReportSignatureReq`. But I renamed it a bit to make it more clear.

Reply:
You're describing [domain driven design](https://en.wikipedia.org/wiki/Domain-driven_design). I haven't read the books but have friends who swear by it.